sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/utils/HttpPipelineProvider.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.utils;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.HttpPolicyProviders;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.management.AuthenticationPolicy;
+import com.azure.management.UserAgentPolicy;
+import com.azure.management.resources.fluentcore.policy.ProviderRegistrationPolicy;
+import com.azure.management.resources.fluentcore.policy.ResourceManagerThrottlingPolicy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A Http Pipeline Provider.
+ */
+public final class HttpPipelineProvider {
+
+    private HttpPipelineProvider() {
+    }
+
+    /**
+     * Creates http pipeline with token credential
+     *
+     * @param credential the token credential
+     * @return the http pipeline
+     */
+    public static HttpPipeline buildHttpPipeline(TokenCredential credential) {
+        // TODO: basic policies should be provided
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        HttpLogOptions httpLogOptions = new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC);
+        policies.add(new UserAgentPolicy(httpLogOptions, null));
+
+        List<HttpPipelinePolicy> retryPolicies = new ArrayList<>();
+        retryPolicies.add(new ProviderRegistrationPolicy(credential));
+        retryPolicies.add(new ResourceManagerThrottlingPolicy());
+        retryPolicies.add(new AuthenticationPolicy(credential, AzureEnvironment.AZURE, null));
+        retryPolicies.add(new HttpLoggingPolicy(httpLogOptions));
+        return buildHttpPipeline(policies, retryPolicies, null);
+    }
+
+    /**
+     * Creates http pipeline with policies and http client.
+     *
+     * @param policies the policies not required in retry strategy
+     * @param retryPolicies the policies required in retry strategy
+     * @param httpClient the http client
+     * @return the http pipeline
+     */
+    public static HttpPipeline buildHttpPipeline(
+        List<HttpPipelinePolicy> policies,
+        List<HttpPipelinePolicy> retryPolicies,
```

Comment:
I think this name is little confused. You even add any `retryPolicy` in pipeline. Not sure we need such function or not.

Reply:
This is mainly about how we handle the `HttpPolicyProviders.addBeforeRetryPolicies` and `HttpPolicyProviders.addAfterRetryPolicies`. We will have policies required to add before RetryPolicies as per my understanding, which is existing behavior before refactoring.

Reply:
The mainly concern is what you want to do by this function. If you want customer to use, I think the pipeline builder will be more clear than such a function. If you only want to use it internally, I think the parameters should be separated.

Reply:
The purpose for this provider is to reuse these pieces of code when creating internal http pipeline. Maybe I should make them private.